### DISCUSSION TITLE
Dependency upgrades on Lerna monorepo started failing

### DISCUSSION THREAD

**Discussion by aisamu** [renovatebot/renovate#18928] [Request Help] [answered]
We have a Lerna monorepo that started (`< 7` days) having issues on all dependency upgrades.

All the Renovate PRs now fail during a standard `npm ci` step:

```
npm ERR! npm ci can only install packages when your package.json and
  package-lock.json or npm-shrinkwrap.json are in sync.
  Please update your lock file with `npm install` before continuing.
```

We found various discussion topics and issues around Lerna and/or lock-files, such as <https://github.com/renovatebot/renovate/discussions/16802> and <https://github.com/renovatebot/renovate/pull/1486>

Guided by those, we checked the lock outcome of running `lerna boostrap`, `npm install` and `npm install --package-lock-only`. Surprisingly, all the resulting lock-files were "correct" - a subsequent `npm ci` worked.

These above results are available on here: <https://github.com/Stedi-Legacy/renovate-lerna>.

On that reproduction repository, there are three pull requests bumping the same dependency across the root and the lerna package with different strategies:

-   One with `npm i` or `lerna bootstrap` (works)
-   One with `npm i --package-lock-only` (works)
-   One by Renovate (fails)

A Github Action runs `npm ci` as part of a required check.

I've tried to find configuration or otherwise relevant project differences between our last working Renovate PR and the failing ones, but there was nothing except version bumps.

On the Renovate PR, we see extra changes on the lock-file of the form:

```diff
@@ -1743,8 +1743,8 @@
       }
     },
     "node_modules/a-package": {
-      "resolved": "packages/a-package",
-      "link": true
+      "version": "0.0.0",
+      "resolved": "file:packages/a-package"
     },
     "node_modules/abbrev": {
       "version": "1.1.1",
@@ -7445,12 +7445,6 @@
       "engines": {
         "node": ">=10"
       }
-    },
-    "packages/a-package": {
-      "version": "0.0.0",
-      "devDependencies": {
-        "constructs": "10.1.159"
-      }
     }
   },
   "dependencies": {
@@ -8860,10 +8854,7 @@
       }
     },
     "a-package": {
-      "version": "file:packages/a-package",
-      "requires": {
-        "constructs": "10.1.159"
-      }
+      "version": "0.0.0"
     },
     "abbrev": {
       "version": "1.1.1",
```

Is there something we could try to get those back in operation?

Happy to provide further information!

**Comment by viceice** [comment, marked as answer]
check the debug logs and past the branch related messages, where npm and lerna is called to update the lockfile

**Reply by rarkins** [reply]
From that you can see the exact node and npm versions used as well as the commands run

**Reply by aisamu** [reply]
Here's an excerpt of the logs containing the information `rarkins` mentions:
<details><summary>Renovate log file excerpt</summary>
<pre>
DEBUG: Matched 1 file(s) for manager github-actions: .github/workflows/check.yml
DEBUG: Matched 2 file(s) for manager npm: package.json, packages/a-package/package.json
DEBUG: npm file package.json has name "renovate-lerna"
DEBUG: Automatically setting skipInstalls to false
DEBUG: npm file packages/a-package/package.json has name "a-package"
DEBUG: Detecting pnpm Workspaces
DEBUG: Detecting Lerna and Yarn Workspaces
DEBUG: Finding locked versions
DEBUG: Found package-lock.json for package.json
DEBUG: Found package-lock.json for packages/a-package/package.json
DEBUG: Found github-actions package files
DEBUG: Found npm package files
DEBUG: Found 3 package file(s)
INFO: Dependency extraction complete
{
  "baseBranch": "main",
  "stats": {
    "managers": {
      "github-actions": {
        "fileCount": 1,
        "depCount": 3
      },
      "npm": {
        "fileCount": 2,
        "depCount": 4
      }
    },
    "total": {
      "fileCount": 3,
      "depCount": 7
    }
  }
}
DEBUG: Release 6.0.3 is pending status checks
{
  "depName": "lerna",
  "check": "stabilityDays"
}
DEBUG: Release 6.0.2 is pending status checks
{
  "depName": "lerna",
  "check": "stabilityDays"
}
DEBUG: Release 10.1.160 is pending status checks
{
  "depName": "constructs",
  "check": "stabilityDays"
}
DEBUG: Release 10.1.159 is pending status checks
{
  "depName": "constructs",
  "check": "stabilityDays"
}
DEBUG: Release 10.1.158 is pending status checks
{
  "depName": "constructs",
  "check": "stabilityDays"
}
DEBUG: Release 10.1.157 is pending status checks
{
  "depName": "constructs",
  "check": "stabilityDays"
}
DEBUG: Release 10.1.156 is pending status checks
{
  "depName": "constructs",
  "check": "stabilityDays"
}
DEBUG: Release 10.1.155 is pending status checks
{
  "depName": "constructs",
  "check": "stabilityDays"
}
DEBUG: Release 10.1.154 is pending status checks
{
  "depName": "constructs",
  "check": "stabilityDays"
}
DEBUG: Release 10.1.153 is pending status checks
{
  "depName": "constructs",
  "check": "stabilityDays"
}
DEBUG: Release 10.1.152 is pending status checks
{
  "depName": "constructs",
  "check": "stabilityDays"
}
DEBUG: Release 10.1.151 is pending status checks
{
  "depName": "constructs",
  "check": "stabilityDays"
}
DEBUG: All releases are pending - using latest
{
  "depName": "constructs",
  "bucket": "non-major"
}
DEBUG: Release 10.1.160 is pending status checks
{
  "depName": "constructs",
  "check": "stabilityDays"
}
DEBUG: Release 10.1.159 is pending status checks
{
  "depName": "constructs",
  "check": "stabilityDays"
}
DEBUG: Release 10.1.158 is pending status checks
{
  "depName": "constructs",
  "check": "stabilityDays"
}
DEBUG: Release 10.1.157 is pending status checks
{
  "depName": "constructs",
  "check": "stabilityDays"
}
DEBUG: Release 10.1.156 is pending status checks
{
  "depName": "constructs",
  "check": "stabilityDays"
}
DEBUG: Release 10.1.155 is pending status checks
{
  "depName": "constructs",
  "check": "stabilityDays"
}
DEBUG: Release 10.1.154 is pending status checks
{
  "depName": "constructs",
  "check": "stabilityDays"
}
DEBUG: Release 10.1.153 is pending status checks
{
  "depName": "constructs",
  "check": "stabilityDays"
}
DEBUG: Release 10.1.152 is pending status checks
{
  "depName": "constructs",
  "check": "stabilityDays"
}
DEBUG: Release 10.1.151 is pending status checks
{
  "depName": "constructs",
  "check": "stabilityDays"
}
DEBUG: All releases are pending - using latest
{
  "depName": "constructs",
  "bucket": "non-major"
}
DEBUG: PackageFiles.add() - Package file saved for base branch
{
  "baseBranch": "main"
}
DEBUG: Package releases lookups complete
{
  "baseBranch": "main"
}
DEBUG: branchifyUpgrades
DEBUG: Using group branchName template
DEBUG: Dependency lerna is part of group lerna monorepo
DEBUG: 5 flattened updates found: actions/checkout, actions/setup-node, lerna, constructs, constructs
DEBUG: Returning 4 branch(es)
DEBUG: config.repoIsOnboarded=true
DEBUG: packageFiles with updates
{
  "baseBranch": "main",
  "config": {
    "github-actions": [
      {
        "packageFile": ".github/workflows/check.yml",
        "deps": [
          {
            "depName": "actions/checkout",
            "commitMessageTopic": "{{{depName}}} action",
            "datasource": "github-tags",
            "versioning": "docker",
            "depType": "action",
            "replaceString": "actions/checkout@e2f20e631ae6d7dd3b768f56a5d2af784dd54791 # tag=v2",
            "autoReplaceStringTemplate": "{{depName}}@{{#if newDigest}}{{newDigest}}{{#if newValue}} # {{newValue}}{{/if}}{{/if}}{{#unless newDigest}}{{newValue}}{{/unless}}",
            "currentValue": "v2",
            "currentDigest": "e2f20e631ae6d7dd3b768f56a5d2af784dd54791",
            "depIndex": 0,
            "updates": [
              {
                "bucket": "major",
                "newVersion": "v3",
                "newValue": "v3",
                "releaseTimestamp": "2022-10-04T09:37:06.000Z",
                "newMajor": 3,
                "newMinor": null,
                "updateType": "major",
                "newDigest": "93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8",
                "branchName": "renovate/actions-checkout-3.x"
              }
            ],
            "warnings": [],
            "sourceUrl": "https://github.com/actions/checkout",
            "currentVersion": "v2",
            "isSingleVersion": true,
            "fixedVersion": "v2"
          },
          {
            "depName": "actions/setup-node",
            "commitMessageTopic": "{{{depName}}} action",
            "datasource": "github-tags",
            "versioning": "docker",
            "depType": "action",
            "replaceString": "actions/setup-node@1f8c6b94b26d0feae1e387ca63ccbdc44d27b561 # tag=v2",
            "autoReplaceStringTemplate": "{{depName}}@{{#if newDigest}}{{newDigest}}{{#if newValue}} # {{newValue}}{{/if}}{{/if}}{{#unless newDigest}}{{newValue}}{{/unless}}",
            "currentValue": "v2",
            "currentDigest": "1f8c6b94b26d0feae1e387ca63ccbdc44d27b561",
            "depIndex": 1,
            "updates": [
              {
                "bucket": "major",
                "newVersion": "v3",
                "newValue": "v3",
                "releaseTimestamp": "2022-10-13T10:11:08.000Z",
                "newMajor": 3,
                "newMinor": null,
                "updateType": "major",
                "newDigest": "8c91899e586c5b171469028077307d293428b516",
                "branchName": "renovate/actions-setup-node-3.x"
              }
            ],
            "warnings": [],
            "sourceUrl": "https://github.com/actions/setup-node",
            "currentVersion": "v2",
            "isSingleVersion": true,
            "fixedVersion": "v2"
          },
          {
            "depName": "volta-cli/action",
            "commitMessageTopic": "{{{depName}}} action",
            "datasource": "github-tags",
            "versioning": "docker",
            "depType": "action",
            "replaceString": "volta-cli/action@25888009cd70dbe17a140f1c56d93f8c09f7bcef # tag=v4",
            "autoReplaceStringTemplate": "{{depName}}@{{#if newDigest}}{{newDigest}}{{#if newValue}} # {{newValue}}{{/if}}{{/if}}{{#unless newDigest}}{{newValue}}{{/unless}}",
            "currentValue": "v4",
            "currentDigest": "25888009cd70dbe17a140f1c56d93f8c09f7bcef",
            "depIndex": 2,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/volta-cli/action",
            "currentVersion": "v4",
            "fixedVersion": "v4"
          }
        ]
      }
    ],
    "npm": [
      {
        "packageFile": "package.json",
        "deps": [
          {
            "depType": "dependencies",
            "depName": "a-package",
            "currentValue": "file:packages/a-package",
            "skipReason": "internal-package",
            "prettyDepType": "dependency",
            "isInternal": true,
            "lockedVersion": null,
            "depIndex": 0,
            "updates": []
          },
          {
            "depType": "devDependencies",
            "depName": "lerna",
            "currentValue": "5.6.2",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": "5.6.2",
            "depIndex": 1,
            "updates": [
              {
                "bucket": "major",
                "newVersion": "6.0.1",
                "newValue": "6.0.1",
                "releaseTimestamp": "2022-10-14T16:48:29.907Z",
                "newMajor": 6,
                "newMinor": 0,
                "updateType": "major",
                "pendingVersions": [
                  "6.0.2",
                  "6.0.3"
                ],
                "branchName": "renovate/major-lerna-monorepo"
              }
            ],
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/lerna/lerna",
            "sourceDirectory": "core/lerna",
            "homepage": "https://lerna.js.org",
            "currentVersion": "5.6.2",
            "isSingleVersion": true,
            "fixedVersion": "5.6.2"
          },
          {
            "depType": "devDependencies",
            "depName": "constructs",
            "currentValue": "10.1.150",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": "10.1.150",
            "depIndex": 2,
            "updates": [
              {
                "bucket": "non-major",
                "newVersion": "10.1.160",
                "newValue": "10.1.160",
                "releaseTimestamp": "2022-11-15T00:14:22.292Z",
                "newMajor": 10,
                "newMinor": 1,
                "updateType": "patch",
                "pendingChecks": true,
                "branchName": "renovate/constructs-10.x"
              }
            ],
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/aws/constructs",
            "currentVersion": "10.1.150",
            "isSingleVersion": true,
            "fixedVersion": "10.1.150"
          }
        ],
        "packageJsonName": "renovate-lerna",
        "packageFileVersion": "0.0.0",
        "npmLock": "package-lock.json",
        "managerData": {
          "lernaJsonFile": "lerna.json",
          "yarnZeroInstall": false,
          "hasPackageManager": false
        },
        "lernaClient": "npm",
        "lernaPackages": [
          "packages/*"
        ],
        "skipInstalls": false,
        "constraints": {},
        "lockFiles": [
          "package-lock.json"
        ]
      },
      {
        "packageFile": "packages/a-package/package.json",
        "deps": [
          {
            "depType": "devDependencies",
            "depName": "constructs",
            "currentValue": "10.1.150",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": "10.1.150",
            "depIndex": 0,
            "updates": [
              {
                "bucket": "non-major",
                "newVersion": "10.1.160",
                "newValue": "10.1.160",
                "releaseTimestamp": "2022-11-15T00:14:22.292Z",
                "newMajor": 10,
                "newMinor": 1,
                "updateType": "patch",
                "pendingChecks": true,
                "branchName": "renovate/constructs-10.x"
              }
            ],
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/aws/constructs",
            "currentVersion": "10.1.150",
            "isSingleVersion": true,
            "fixedVersion": "10.1.150"
          }
        ],
        "packageJsonName": "a-package",
        "packageFileVersion": "0.0.0",
        "npmLock": "package-lock.json",
        "managerData": {
          "lernaJsonFile": "lerna.json",
          "yarnZeroInstall": false,
          "hasPackageManager": false
        },
        "lernaClient": "npm",
        "skipInstalls": false,
        "constraints": {},
        "lockFiles": [
          "package-lock.json"
        ]
      }
    ]
  }
}
DEBUG: processRepo()
DEBUG: Processing 4 branches: renovate/actions-checkout-3.x, renovate/actions-setup-node-3.x, renovate/constructs-10.x, renovate/major-lerna-monorepo
DEBUG: Calculating hourly PRs remaining
DEBUG: getPrList success
{
  "pullsTotal": 2,
  "requestsTotal": 1,
  "apiQuotaAffected": true
}
DEBUG: currentHourStart=2022-11-15T18:00:00.000+00:00
DEBUG: PR hourly limit remaining: 2
DEBUG: Calculating prConcurrentLimit (10)
DEBUG: getBranchPr(renovate/constructs-10.x)
DEBUG: findPr(renovate/constructs-10.x, undefined, open)
DEBUG: Found PR #7
DEBUG: getBranchPr(renovate/actions-checkout-3.x)
DEBUG: findPr(renovate/actions-checkout-3.x, undefined, open)
DEBUG: findPr(renovate/actions-checkout-3.x, undefined, closed)
DEBUG: getBranchPr(renovate/actions-setup-node-3.x)
DEBUG: findPr(renovate/actions-setup-node-3.x, undefined, open)
DEBUG: findPr(renovate/actions-setup-node-3.x, undefined, closed)
DEBUG: getBranchPr(renovate/major-lerna-monorepo)
DEBUG: findPr(renovate/major-lerna-monorepo, undefined, open)
DEBUG: findPr(renovate/major-lerna-monorepo, undefined, closed)
DEBUG: 1 PRs are currently open
DEBUG: PR concurrent limit remaining: 9
DEBUG: Calculated maximum PRs remaining this run: 2
DEBUG: PullRequests limit = 2
DEBUG: Calculating hourly PRs remaining
DEBUG: currentHourStart=2022-11-15T18:00:00.000+00:00
DEBUG: PR hourly limit remaining: 2
DEBUG: Calculating branchConcurrentLimit (10)
DEBUG: 1 already existing branches found: renovate/constructs-10.x
DEBUG: Branch concurrent limit remaining: 9
DEBUG: Calculated maximum branches remaining this run: 2
DEBUG: Branches limit = 2
DEBUG: syncBranchState()(branch="renovate/constructs-10.x")
DEBUG: branch.isUpToDate(): using cached result "true"(branch="renovate/constructs-10.x")
DEBUG: getBranchPr(renovate/constructs-10.x)(branch="renovate/constructs-10.x")
DEBUG: findPr(renovate/constructs-10.x, undefined, open)(branch="renovate/constructs-10.x")
DEBUG: Found PR #7(branch="renovate/constructs-10.x")
DEBUG: branchExists=true(branch="renovate/constructs-10.x")
DEBUG: dependencyDashboardCheck=rebase(branch="renovate/constructs-10.x")
DEBUG: PR rebase requested=false(branch="renovate/constructs-10.x")
DEBUG: Checking if PR has been edited(branch="renovate/constructs-10.x")
DEBUG: branch.isModified(): using cached result "false"(branch="renovate/constructs-10.x")
DEBUG: Found existing branch PR(branch="renovate/constructs-10.x")
DEBUG: Checking schedule(at any time, null)(branch="renovate/constructs-10.x")
DEBUG: No schedule defined(branch="renovate/constructs-10.x")
DEBUG: Update has not passed stability days(branch="renovate/constructs-10.x")
{
  "depName": "constructs",
  "daysElapsed": 0,
  "stabilityDays": 21
}
DEBUG: Update has not passed stability days(branch="renovate/constructs-10.x")
{
  "depName": "constructs",
  "daysElapsed": 0,
  "stabilityDays": 21
}
DEBUG: Manual rebase requested via Dependency Dashboard(branch="renovate/constructs-10.x")
DEBUG: Using reuseExistingBranch: false(branch="renovate/constructs-10.x")
DEBUG: Setting current branch to main(branch="renovate/constructs-10.x")
DEBUG: latest commit(branch="renovate/constructs-10.x")
{
  "branchName": "main",
  "latestCommitDate": "2022-11-14T18:35:09-03:00"
}
DEBUG: manager.getUpdatedPackageFiles() reuseExistingBranch=false(branch="renovate/constructs-10.x")
DEBUG: npm.updateDependency(): devDependencies.constructs = 10.1.160(branch="renovate/constructs-10.x")
DEBUG: Updating constructs in packages/a-package/package.json(branch="renovate/constructs-10.x")
DEBUG: npm.updateDependency(): devDependencies.constructs = 10.1.160(branch="renovate/constructs-10.x")
DEBUG: Updating constructs in package.json(branch="renovate/constructs-10.x")
DEBUG: Updated 2 package files(branch="renovate/constructs-10.x")
DEBUG: Getting updated lock files(branch="renovate/constructs-10.x")
DEBUG: packages/a-package/package.json has lerna lock file(branch="renovate/constructs-10.x")
DEBUG: package.json has lerna lock file(branch="renovate/constructs-10.x")
DEBUG: Writing package.json files(branch="renovate/constructs-10.x")
{
  "packageFiles": [
    "package.json",
    "packages/a-package/package.json"
  ]
}
DEBUG: Writing package-lock.json(branch="renovate/constructs-10.x")
DEBUG: Massaging npm lock file before writing to disk(branch="renovate/constructs-10.x")
DEBUG: Writing package-lock.json(branch="renovate/constructs-10.x")
DEBUG: Massaging npm lock file before writing to disk(branch="renovate/constructs-10.x")
DEBUG: Writing any updated package files(branch="renovate/constructs-10.x")
DEBUG: Writing packages/a-package/package.json(branch="renovate/constructs-10.x")
DEBUG: Writing package.json(branch="renovate/constructs-10.x")
DEBUG: Finding package.json for lerna location "lerna.json"(branch="renovate/constructs-10.x")
DEBUG: npmrc file found in repository(branch="renovate/constructs-10.x")
DEBUG: Writing updated .npmrc file to .npmrc(branch="renovate/constructs-10.x")
DEBUG: Spawning lerna with npm to create lock files(branch="renovate/constructs-10.x")
DEBUG: No node constraint found - using latest(branch="renovate/constructs-10.x")
DEBUG: Using lerna version 5.6.2(branch="renovate/constructs-10.x")
DEBUG: Setting CONTAINERBASE_CACHE_DIR to /tmp/containerbase(branch="renovate/constructs-10.x")
DEBUG: Using docker to execute image: sidecar(branch="renovate/constructs-10.x")
DEBUG: Resolved stable matching version(branch="renovate/constructs-10.x")
{
  "toolName": "node",
  "constraint": null,
  "resolvedVersion": "v18.12.1"
}
DEBUG: Resolved stable matching version(branch="renovate/constructs-10.x")
{
  "toolName": "npm",
  "resolvedVersion": "9.1.1"
}
DEBUG: containerbaseDir is separate from cacheDir(branch="renovate/constructs-10.x")
DEBUG: No tag or tagConstraint specified for image: docker.io/renovate/sidecar(branch="renovate/constructs-10.x")
DEBUG: Docker image is already prefetched: docker.io/renovate/sidecar@sha256:5103d041fbc1e3deb7fd201ad0ed6ade56dc5c7f44c1613b094e58c022167fe8(branch="renovate/constructs-10.x")
DEBUG: exec completed(branch="renovate/constructs-10.x")
{
  "cmd": "docker run --rm --name=renovate_sidecar --label=renovate_child -v \"/mnt/renovate/gh/Stedi-Legacy/renovate-lerna\":\"/mnt/renovate/gh/Stedi-Legacy/renovate-lerna\" -v \"/tmp/renovate-cache\":\"/tmp/renovate-cache\" -v \"/tmp/containerbase\":\"/tmp/containerbase\" -e NPM_CONFIG_CACHE -e npm_config_store -e BUILDPACK_CACHE_DIR -e CONTAINERBASE_CACHE_DIR -w \"/mnt/renovate/gh/Stedi-Legacy/renovate-lerna\" docker.io/renovate/sidecar bash -l -c \"install-tool node v18.12.1 && install-tool npm 9.1.1 && hash -d npm 2>/dev/null || true && install-tool lerna 5.6.2 && lerna info || echo \\\"Ignoring lerna info failure\\\" && npm install --ignore-scripts  --no-audit && lerna bootstrap --no-ci --ignore-scripts -- --ignore-scripts  --no-audit\"",
  "durationMs": 109879,
  "stdout": "installing v2 tool node v18.12.1\nlinking tool node v18.12.1\nnode: v18.12.1 /usr/local/bin/node\nnpm: 8.19.2  /usr/local/bin/npm\nInstalled v2 /usr/local/buildpack/tools/v2/node.sh in 7 seconds\nskip cleanup, not a docker build: ec06aa7b2a8e\ninstalling v2 tool npm v9.1.1\n\nadded 1 package in 7s\nlinking tool npm v9.1.1\n9.1.1\nInstalled v2 /usr/local/buildpack/tools/v2/npm.sh in 9 seconds\nskip cleanup, not a docker build: ec06aa7b2a8e\ninstalling v2 tool lerna v5.6.2\nnpm WARN deprecated readdir-scoped-modules@1.1.0: This functionality has been moved to @npmcli/fs\nnpm WARN deprecated @npmcli/move-file@2.0.1: This functionality has been moved to @npmcli/fs\n\nadded 616 packages in 45s\nlinking tool lerna v5.6.2\n5.6.2\nInstalled v2 /usr/local/buildpack/tools/v2/lerna.sh in 48 seconds\nskip cleanup, not a docker build: ec06aa7b2a8e\n\n Environment info:\n\n  System:\n    OS: Linux 5.4 Ubuntu 20.04.5 LTS (Focal Fossa)\n    CPU: (1) x64 Intel(R) Xeon(R) CPU E5-2670 v2 @ 2.50GHz\n  Binaries:\n    Node: 18.12.1 - /usr/local/bin/node\n    npm: 9.1.1 - /usr/local/bin/npm\n  Utilities:\n    Git: 2.38.1 - /usr/bin/git\n\n\nadded 618 packages in 32s\n\n60 packages are looking for funding\n  run `npm fund` for details\n\nup to date in 4s\n\n60 packages are looking for funding\n  run `npm fund` for details\n",
  "stderr": "lerna notice cli v5.6.2\nnpm WARN deprecated readdir-scoped-modules@1.1.0: This functionality has been moved to @npmcli/fs\nnpm WARN deprecated @npmcli/move-file@2.0.1: This functionality has been moved to @npmcli/fs\ninfo cli using local version of lerna\nlerna notice cli v5.6.2\nlerna info bootstrap root only\n"
}
DEBUG: File is updated: package-lock.json(branch="renovate/constructs-10.x")
DEBUG: File is updated: package-lock.json(branch="renovate/constructs-10.x")
DEBUG: Updated 2 lock files(branch="renovate/constructs-10.x")
{
  "updatedArtifacts": [
    "package-lock.json",
    "package-lock.json"
  ]
}
DEBUG: Getting comments for #7(branch="renovate/constructs-10.x")
DEBUG: Found 0 comments(branch="renovate/constructs-10.x")
DEBUG: isBranchConflicted(main, renovate/constructs-10.x)(branch="renovate/constructs-10.x")
DEBUG: branch.isConflicted(): using cached result "false"(branch="renovate/constructs-10.x")
DEBUG: 3 file(s) to commit(branch="renovate/constructs-10.x")
DEBUG: Preparing files for committing to branch renovate/constructs-10.x(branch="renovate/constructs-10.x")
DEBUG: Setting git author name: Renovate Bot(branch="renovate/constructs-10.x")
DEBUG: Setting git author email: bot@renovateapp.com(branch="renovate/constructs-10.x")
DEBUG: git commit(branch="renovate/constructs-10.x")
{
  "deletedFiles": [],
  "ignoredFiles": [],
  "result": {
    "author": null,
    "branch": "renovate/constructs-10.x",
    "commit": "a601e9478fdd5d838db7206909ac7bcc31e6fbe0",
    "root": false,
    "summary": {
      "changes": 3,
      "insertions": 12,
      "deletions": 21
    }
  }
}
DEBUG: resetToCommit(ac0d375c129853b7b3a967eee065236f64988777)(branch="renovate/constructs-10.x")
</pre></details>

```
DEBUG: Using lerna version 5.6.2(branch="renovate/constructs-10.x")
...
  "toolName": "node",
  "resolvedVersion": "v18.12.1"
...
  "toolName": "npm",
  "resolvedVersion": "9.1.1"
```

**Reply by viceice** [reply]
it seems you don't have any constraints on node or npm, so renovate will always latest node lts and latest npm.

so this will probably cause the difference.

**Reply by viceice** [reply]
```sh
> npm install --ignore-scripts  --no-audit
> lerna bootstrap --no-ci --ignore-scripts -- --ignore-scripts  --no-audit
```

this is the command renovate runs on your repo.

**Reply by aisamu** [reply]
Aha! 🥳 
Mistery solved, thank you!
We were only pinning Node to the latest LTS and using the bundled npm version:

> Latest LTS Version: 18.12.1 (includes npm 8.19.2)

`npm@latest` is 9, which introduces those differences.

**Reply by viceice** [reply]
yes, npm v9.1.1 was just tagged as latest
